R/mouse/mouse_deconvolution_methods.R

```diff
@@ -0,0 +1,228 @@
+                                   package = 'immunedeconv', mustWork=TRUE)
+  source('./seqImmuCC_LLSR.R')
+  
+  if (method == 'SVR'){
+    
+    assert("CIBERSORT.R is provided", exists("cibersort_binary", envir=config_env))
+    source(get("cibersort_binary", envir=config_env))
+    
+    temp.expression.file = tempfile()
+    
+    write_tsv(as_tibble(gene.expression.matrix, rownames = 'gene_symbol'), path = temp.expression.file)
+    perm = 100
+    results = CIBERSORT(signature.path, temp.expression.file, perm, QN=FALSE)
+    results = results %>%
+     .[, !colnames(.) %in% c("RMSE", "P-value", "Correlation")]
+    
+  } else if (method == 'LLSR'){
+    
+    signature = read.table(signature.path, header=T, sep="\t", row.names=1, check.names=F)
+    results = LLSR(signature, gene.expression.matrix)
+  }
+  
+  
+  return(results)
+}
+
+#' Deconvolute using DCQ
+#' 
+#' @param gene.expression.matrix a m x n matrix with m genes and n samples
+#' @param ref.samples a vector containing the names of the samples to be 
+#'    used to standardize the expressions. If NULL, all the samples are used. 
+#' @param n.repeats the number of models to generate. Predicted cell quantities
+#'    will then be averaged.    
+#' @export
+#'      
+deconvolute_DCQ = function(gene.expression.matrix, 
+                           ref.samples = NULL, n.repeats = 10){
+  
+  # We need to scale the counts
+  gene.expression.matrix = as.matrix(gene.expression.matrix)
+  
+  if (is.null(ref.samples)){
+    rows.means = rowMeans(gene.expression.matrix)
+    rows.SD = rowSds(gene.expression.matrix)
+  } else {
+    rows.means = rowMeans(gene.expression.matrix[, samples])
+    rows.SD = rowSds(gene.expression.matrix[, samples])
+  }
+  rows.SD[rows.SD==0] = 1
+  
+  gene.expression.matrix = (gene.expression.matrix - rows.means)/rows.SD
+  
+  arguments = dots_list(reference_data = immgen_dat, 
+                         mix_data = gene.expression.matrix, 
+                         marker_set = DCQ_mar, 
+                         number_of_repeats = n.repeats)
+  call = rlang::call2(dcq, !!!arguments)
+  results = eval(call)
+  return(results$average)
+}
+
+#' Deconvolute using BASE
+#' 
+#' @param gene.expression.matrix a m x n matrix with m genes and n samples. Data
+#'    should be TPM normalized and log10 scaled.
+#' @param n.permutations the number of permutations of each sample expression
+#'    to generate. These are used to normalize the results. 
+#' @param log10 logical. if TRUE, log10 transforms the expression matrix. 
+#' 
+#' @export
+#' 
+deconvolute_BASE = function(gene.expression.matrix, n.permutations = 100, log10 = TRUE){
+  
+  base.compendium.path <- system.file('extdata', 'mouse_deconvolution', 'BASE_immune_compendium.rds',
+                                      package = 'immunedeconv', mustWork=TRUE)
+  immune.compendium = load(base.compendium.path)
+  
+  source('./BASE.R')
```

Comment:
There should be no need to use `source`. In a package you can use all exported from other files.